components/movementsensor/gpsutils/vrs.go

```diff
@@ -36,7 +39,7 @@ func ConnectToVirtualBase(ntripInfo *NtripInfo,
 
 	// Construct HTTP headers with CRLF line endings
 	httpHeaders := "GET " + mp + " HTTP/1.1\r\n" +
-		"Host: " + serverAddr + "\r\n" +
+		"Host: " + serverAddr.Host + "\r\n" +
```

Comment:
I have no idea if this is the right way to use this package or what we're really looking for here? let me know if you have any familiarity with this url package

Reply:
So I believe this is correct to use the Host field (as it includes the port), however, this whole function still likely needs to return an error when something goes wrong. Right now, it just returns nil (in possibly two different places above) and that causes a panic when the calling code tries to use the returned value as a connection.